src/instance/api/dom.js

```diff
@@ -108,6 +108,8 @@ export default function (Vue) {
     // for transitions
     if (!inDocument) withTransition = false
     var self = this
+    // if the Instance contains a beforeDetached event trigger it
+    if (self.$options.beforeDetached) self._callHook('beforeDetached')
```

Comment:
no need for the if check here - `_callHook` would be a no-op if the hook does not exist.